dev/com.ibm.ws.kernel.boot_fat/fat/src/com/ibm/ws/kernel/boot/VerboseLogTest.java

```diff
@@ -0,0 +1,205 @@
+import componenttest.topology.impl.LibertyServer;
+import componenttest.topology.impl.LibertyServerFactory;
+
+@RunWith(FATRunner.class)
+public class VerboseLogTest {
+    private static final Class<?> c = VerboseLogTest.class;
+
+    @Rule
+    public TestName testName = new TestName();
+
+    private static LibertyServer server;
+    private final static String SERVER_NAME = "com.ibm.ws.kernel.boot.verbose.fat";
+    static String executionDir;
+    static File verboseLog;
+    static File verboseLogServerRoot;
+    static String serverCommand;
+    static File jvmoptionsserverroot;
+
+    @BeforeClass
+    public static void before() throws Exception {
+        server = LibertyServerFactory.getLibertyServer(SERVER_NAME);
+        executionDir = server.getInstallRoot();
+        verboseLog = new File(executionDir + "/usr/servers/" + SERVER_NAME + "/logs/verbosegc.001.log");
+        verboseLogServerRoot = new File(executionDir + "/usr/servers/" + SERVER_NAME + "/verbosegc.001.log");
+        jvmoptionsserverroot = new File(executionDir + "/usr/servers/" + SERVER_NAME + "/jvm.options");
+        if (server.getMachine().getOperatingSystem() == OperatingSystem.WINDOWS)
+            serverCommand = "bin\\server.bat";
+        else
+            serverCommand = "bin/server";
+    }
+
+    @AfterClass
+    public static void after() throws Exception {
+        if (server.isStarted()) {
+            server.stopServer();
+        }
+    }
+
+    @Test
+    public void testDefaultVerboseLogging() throws Exception {
+        // Test with no jvm.options, verbose log should appear by default
+        Log.entering(c, testName.getMethodName());
+
+        delJvmOptions();
+
+        String[] parms = new String[2];
+        parms[0] = "start";
+        parms[1] = SERVER_NAME;
+
+        Properties envVars = new Properties();
+        envVars.put("CDPATH", ".");
+
+        ProgramOutput po = server.getMachine().execute(serverCommand, parms, executionDir, envVars);
+
+        Log.info(c, testName.getMethodName(), "server start stdout = " + po.getStdout());
+        Log.info(c, testName.getMethodName(), "server start stderr = " + po.getStderr());
+
+        server.waitForStringInLog("CWWKF0011I");
+        server.resetStarted();
+
+        assertTrue("the server should have been started", server.isStarted());
+        assertTrue("verbosegc log should have been created", verboseLog.exists());
+
+        server.stopServer();
+    }
+
+    @Test
+    public void testJvmTurnOfVerbose() throws Exception {
+        // Test with jvm.options to turn off verbose log, no verbose log should appear
+        Log.entering(c, testName.getMethodName());
+
+        delJvmOptions();
+
+        String[] parms = new String[2];
+        parms[0] = "start";
+        parms[1] = SERVER_NAME;
+
+        Writer isw = new OutputStreamWriter(new FileOutputStream(jvmoptionsserverroot), "UTF-8");
+        BufferedWriter bw = new BufferedWriter(isw);
+        bw.write("-Dverbosegc=false\n");
```

Comment:
`-D` is a JVM system property. I think you'll need to write `VERBOSEGC=false` to `server.env` instead of `jvm.options` and then also check for this in the `server` scripts.

Reply:
@rsherget Just double checking if this is handled?